### PR TITLE
Add indices for add to consult

### LINKED ISSUE #170: Make AddToConsultCommand accept indices
Currently, the AddToConsultCommand adds students to a consult by names

Sample of current command to add 2 students, one named John Doe, another names Harry Ng:
`addtoconsult 1 n/John Doe n/Harry Ng`

We want to change it to accept indices of students too

Sample command after change to add 2 students, one named John Doe and the second student in the currently filtered list:
`addtoconsult 1 n/John Doe i/2`

### PR DESCRIPTION
**Pull request by Clarenceeey**
Fixes #170 

Added support for passing indices when adding students.

Sample usage:
`addtoconsult 1 n/Harry Ng n/John Doe i/1 i/2`

This command adds students at index 1 and 2 and Harry Ng and John Doe students to first index of the filtered consultation list

See #170 for more details

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/consultation/AddToConsultCommand.java [hunk 1]
```diff
@@ -65,12 +81,44 @@ public CommandResult execute(Model model) throws CommandException {
 
         for (Name studentName : studentNames) {
             Student student = model.findStudentByName(studentName)
-                    .orElseThrow(() -> new CommandException("Student not found: " + studentName));
+                    .orElseThrow(() -> new CommandException(
+                            String.format(MESSAGE_STUDENT_NOT_FOUND, studentName)));
```
Comment (yhanyi) [inline src/main/java/seedu/address/logic/commands/consultation/AddToConsultCommand.java:85]: Nice use of message.

File: src/main/java/seedu/address/logic/parser/consultation/AddToConsultCommandParser.java [hunk 2]
```diff
@@ -41,7 +42,7 @@ public AddToConsultCommand parse(String args) throws ParseException {
                 AddToConsultCommand.MESSAGE_USAGE), pe);
         }
 
-        if (argMultimap.getAllValues(PREFIX_NAME).isEmpty()) {
+        if (argMultimap.getAllValues(PREFIX_NAME).isEmpty() && argMultimap.getAllValues(PREFIX_INDEX).isEmpty()) {
```
Comment (yhanyi) [inline src/main/java/seedu/address/logic/parser/consultation/AddToConsultCommandParser.java:45]: Good use of argMultimap.

**Review by yhanyi — COMMENTED** [review]
LGTM.

File: src/main/java/seedu/address/logic/commands/consultation/AddToConsultCommand.java [hunk 3]
```diff
@@ -33,27 +36,40 @@ public class AddToConsultCommand extends Command {
             + "Example: " + COMMAND_WORD + " 1 n/John Doe n/Harry Ng";
 
     public static final String MESSAGE_ADD_TO_CONSULT_SUCCESS = "Added students to the Consultation: %1$s";
-    public static final String MESSAGE_DUPLICATE_STUDENT_IN_CONSULTATION = "%s is already added to the consultation!";
+
+    public static final String MESSAGE_STUDENT_NOT_FOUND = "Student(s) %s not found";
+    public static final String MESSAGE_DUPLICATE_STUDENT_IN_CONSULTATION_BY_NAME =
+            "%s is already added to the consultation!";
+    public static final String MESSAGE_DUPLICATE_STUDENT_IN_CONSULTATION_BY_INDEX =
```
Comment (notnotmax) [inline src/main/java/seedu/address/logic/commands/consultation/AddToConsultCommand.java:43]: I like the distinction between the name and the index error message. Helps the user determine which part of their command is invalid.

File: src/main/java/seedu/address/logic/commands/consultation/AddToConsultCommand.java [hunk 4]
```diff
@@ -33,27 +36,40 @@ public class AddToConsultCommand extends Command {
...
+    private final List<Index> indices;
 
     /**
      * @param index of the consultation in the filtered consultation list
      * @param studentNames list of student names to add to the consultation
+     * @param indices list of indices of students in the current filtered list to add to the current consultation
      */
-    public AddToConsultCommand(Index index, List<Name> studentNames) {
+    public AddToConsultCommand(Index index, List<Name> studentNames, List<Index> indices) {
         requireNonNull(index);
```
Comment (notnotmax) [inline src/main/java/seedu/address/logic/commands/consultation/AddToConsultCommand.java:58]: Non-issue, but a `requireAllNonNull` could also work here.

File: src/test/java/seedu/address/logic/commands/consultation/AddToConsultCommandTest.java [hunk 5]
```diff
@@ -1,122 +1,234 @@
...
+            new Name(CARL.getName().fullName), new Name(BOB.getName().fullName));
+    private final ObservableList<Name> validStudentNames = FXCollections.observableArrayList(
+            new Name(CARL.getName().fullName), new Name(DANIEL.getName().fullName));
+    private final ObservableList<Name> duplicateStudentNames = FXCollections.observableArrayList(
+            new Name(CARL.getName().fullName), new Name(CARL.getName().fullName));
+    private final ObservableList<Index> validStudentIndices = FXCollections.observableArrayList(
+            INDEX_FIRST_STUDENT, INDEX_SECOND_STUDENT); //contains Alice and Bob
+    private ObservableList<Index> invalidIndices;
+
+    @BeforeEach
```
Comment (notnotmax) [inline src/test/java/seedu/address/logic/commands/consultation/AddToConsultCommandTest.java:56]: I didn't know this was a thing. Nice use of it.

**Review by notnotmax — APPROVED** [review]
lgtm, ready to merge.

**Pull request merged**